src/snowflake/snowpark/mock/mock_plan.py

```diff
@@ -317,14 +329,23 @@ def execute_mock_plan(
                     child_rf.insert(
                         len(child_rf.columns),
                         intermediate_mapped_column[i],
-                        ColumnEmulator(data=[agg_expr.child.value] * len(child_rf)),
+                        ColumnEmulator(
+                            data=[agg_expr.child.value] * len(child_rf),
+                            sf_type=ColumnType(
+                                agg_expr.child.datatype, agg_expr.child.nullable
+                            ),
+                        ),
                     )
                 elif isinstance(agg_expr.child, (ListAgg, FunctionExpression)):
                     # function expression will be evaluated later
                     child_rf.insert(
                         len(child_rf.columns),
                         intermediate_mapped_column[i],
-                        ColumnEmulator(data=[None] * len(child_rf), dtype=object),
+                        ColumnEmulator(
+                            data=[None] * len(child_rf),
+                            dtype=object,
+                            sf_type=ColumnType(ArrayType(), True),
```

Comment:
I think we can remove the sf_type here, the sf_type will be set after evaluation of the `agg_expr.child`, column added here served as a place holder.